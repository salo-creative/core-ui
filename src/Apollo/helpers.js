import { get, hasIn, forEach, isEmpty, find } from 'lodash';

// HELPERS & CONSTANTS
export const isLoading = ({ networkStatus, skip = false }) => !skip && (
  // Initial load
  (networkStatus === 1)
  // Reload due to variables change
  || (networkStatus === 2)
  // Reload due to refetch
  || (networkStatus === 4)
);

export const isError = ({ networkStatus, error, skip = false }) => (
  error || (!skip && (networkStatus === 8)) // Network error (exclude skip as that also causes status of 8)
);

export const getRequestStatus = ({ networkStatus, error, skip = false }) => {
  return {
    requestLoading: isLoading({ networkStatus, skip }),
    requestError: isError({ networkStatus, error, skip })
  };
};

export const parseApolloError = ({ error, propName = 'APOLLO_ERROR' }) => {
  let code = 400;
  let message = 'Something went wrong!';
  const errors = [];
  const extra = {};
  let name = propName; // For custom sentry logging
  if (error.graphQLErrors) {
    // Find custom errors if they exist
    const customError = find(error.graphQLErrors, { message: 'SC_ERROR' });

    if (hasIn(customError, 'path')) {
      extra.path = customError.path.join(' -> ');
      name = propName === 'APOLLO_ERROR' ? `APOLLO_ERROR: ${ customError.path[0] }` : propName;
    }

    code = get(customError, 'extensions.status', code);

    // Handle custom error message
    const serverMessage = get(customError, 'extensions.message');
    if (serverMessage && typeof serverMessage === 'string') {
      message = serverMessage;
    } else if (typeof serverMessage === 'object') {
      message = serverMessage.message; // eslint-disable-line
      code = serverMessage.name === 'ValidationError' ? 422 : 400;
    }

    // Handle custom errors array
    const serverErrorsMessages = get(customError, 'extensions.errors');
    if (!isEmpty(serverErrorsMessages)) {
      forEach(serverErrorsMessages, errMessage => {
        if (typeof errMessage === 'string') errors.push(errMessage);
      });
    }
  }

  return {
    code,
    errors,
    extra,
    message,
    name
  };
};