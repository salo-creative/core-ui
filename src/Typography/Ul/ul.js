import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import FluidType from '../../helpers/fluidType';

const StyledUl = styled.ul`
  font-weight: 400;
  width: 100%;
  color: ${ ({ color }) => color };
  margin: ${ ({ margin }) => margin };
  padding: ${ ({ padding }) => padding };

  ${ ({ fontSize, lineHeight, minFont, maxFont, minLine, maxLine }) => {
    if (fontSize === '') {
      return `${ FluidType({ fontSize: [minFont, maxFont], lineHeight: [minLine, maxLine] }) }`;
    }
    return `
      font-size: ${ fontSize };
      line-height: ${ lineHeight };
    `;
  } }
  li p {
    margin: 0;
  }
`;

const Ul = ({ children, ...props }) => {
  return <StyledUl { ...props }>{ children }</StyledUl>;
};

Ul.propTypes = {
  color: PropTypes.string,
  fontSize: PropTypes.string,
  lineHeight: PropTypes.string,
  margin: PropTypes.string,
  padding: PropTypes.string,
  minFont: PropTypes.number,
  minLine: PropTypes.number,
  maxFont: PropTypes.number,
  maxLine: PropTypes.number
};

Ul.defaultProps = {
  color: '#262729',
  fontSize: '',
  lineHeight: '',
  margin: '0 0 20px',
  padding: '0 0 0 25px',
  minFont: 14,
  maxFont: 17,
  minLine: 17,
  maxLine: 29
};

Ul.displayName = 'Ul';

export default Ul;