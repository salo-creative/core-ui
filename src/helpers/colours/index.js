import { darken } from 'polished';

/**
 * COLOUR DEFINITIONS
 *
 * Fixed colour definitions to ensure
 * centralised definition of consts
 */
export const colours = {
  blue: '#00ACED',
  brightBlue: '#00D2FF',
  charcoal: '#333333',
  darkGrey: '#6c757d',
  green: '#31D238',
  grey: '#e9ecef',
  orange: '#f47a3e',
  paleGrey: '#fafcff',
  pink: '#EF358A',
  purple: '#62358A',
  red: '#f15b60',
  yellow: '#FFAE00'
};

export const saloTheme = {
  primary: colours.blue,
  secondary: colours.pink,
  success: colours.green,
  error: colours.red,
  warning: colours.yellow,
  info: colours.grey,
  font: colours.charcoal,
  fontInverse: '#fff',
  bodyFont: '\'Brother\', sans-serif',
  headerFont: '\'Brother\', sans-serif',
  ...colours
};

/**
 * BOX SHADOW
 *
 * Centralised definition for shadows
 * @param {*} size - string value for the size. [default | medium, | large]
 * @returns box shadow css property string
 */
export const boxShadow = (size) => {
  switch (size) {
    case 'large':
    case 'small':
    case 'default':
    default:
      return '2px 2px 2px 0px rgba(0,0,0,0.2)';
  }
};

/**
 * BUTTON THEMES
 *
 * Definition for the button themes from the predefined colours
 */
export const buttonThemes = ({ primary, secondary, font, fontInverse, success, error }) => ({
  white: {
    border: '#fff',
    background: '#fff',
    color: font,
    borderHov: colours.paleGrey,
    backgroundHov: colours.paleGrey,
    colorHov: font,
    shadow: true
  },
  clear: {
    border: font,
    background: 'transparent',
    color: font,
    borderHov: font,
    backgroundHov: 'rgba(0,0,0,0.1)',
    colorHov: font,
    shadow: false
  },
  clear_inverse: {
    border: fontInverse,
    background: 'transparent',
    color: fontInverse,
    borderHov: darken(0.3, fontInverse),
    backgroundHov: 'transparent',
    colorHov: darken(0.3, fontInverse),
    shadow: false
  },
  text: {
    border: 'transparent',
    background: 'transparent',
    color: colours.darkGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: font,
    shadow: false
  },
  text_inverse: {
    border: 'transparent',
    background: 'transparent',
    color: colours.paleGrey,
    borderHov: 'transparent',
    backgroundHov: 'transparent',
    colorHov: fontInverse,
    shadow: false
  },
  primary: {
    border: primary,
    background: primary,
    color: fontInverse,
    borderHov: darken(0.15, primary),
    backgroundHov: darken(0.15, primary),
    colorHov: fontInverse,
    shadow: true
  },
  secondary: {
    border: secondary,
    background: secondary,
    color: fontInverse,
    borderHov: darken(0.15, secondary),
    backgroundHov: darken(0.15, secondary),
    colorHov: fontInverse,
    shadow: true
  },
  error: {
    border: error,
    background: error,
    color: fontInverse,
    borderHov: darken(0.15, error),
    backgroundHov: darken(0.15, error),
    colorHov: fontInverse,
    shadow: true
  },
  success: {
    border: success,
    background: success,
    color: fontInverse,
    borderHov: darken(0.15, success),
    backgroundHov: darken(0.15, success),
    colorHov: fontInverse,
    shadow: true
  }
});