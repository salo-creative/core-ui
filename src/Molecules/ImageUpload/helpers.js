import { dataURItoBlob } from '../../helpers/files';

export const onImageDrop = async ({
  files,
  maxFileSize,
  onUpload,
  rejected,
  setFileSizeError
}) => {
  if (rejected.length > 0) {
    if (rejected[0].size >= maxFileSize) {
      setFileSizeError(true);
    }
    return;
  }
  
  if (!files.length) return;
  
  setFileSizeError(false);
  
  const file = files[0];
  const img = new Image();
  
  img.onload = (e) => {
    const sizes = {
      width: e.target.width,
      height: e.target.height
    };
  
    const reader = new FileReader();
  
    reader.onload = (event) => {
      // Image service requires a binary, not base64 so we need to convert back
      const fileBlob = dataURItoBlob(event.target.result);
      onUpload({
        blob: fileBlob, base64: event.target.result, sizes
      });
    };
    reader.readAsDataURL(file);
  };
  
  img.src = URL.createObjectURL(file);
};