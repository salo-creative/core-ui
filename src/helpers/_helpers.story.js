import breakpoints from './breakpoints/_breakpoints.story';
import colours from './colours/_colours.story';
import environments from './environments/_environments.story';