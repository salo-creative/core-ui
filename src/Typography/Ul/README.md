# Unordered list

Ul is a pure styled component render. Renders a standard `<ul>` with themed styling

-------

## Usage

Install

```javascript
yarn add @salo/core-ui
```

Include

```javascript
import { Ul } from '@salo/core-ui/Typography/ul';
```

Implement as follows

```javascript
<Ul>
  // My content here
</Ul>
```

### For full prop types and usage see storybook info/knobs