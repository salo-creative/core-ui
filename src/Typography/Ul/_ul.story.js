import React from 'react';

// load tests
import { withTests } from '@storybook/addon-jest';
import { withKnobs, text, color } from '@storybook/addon-knobs';
import results from '../../../.storybook/jest-test-results.json';

// FEATURED COMPONENT
import { Ul } from '../../index';

// README //
import README from './README.md';

// Start of story logic
export const Basic = () => {
  const fontSize = text('Font size', '');
  const lineHeight = text('Line height', '');
  const colour = color('Colour', '#262729');
  const margin = text('Margin', '0 0 20px');
  return (
    <Ul color={ colour } fontSize={ fontSize } lineHeight={ lineHeight } margin={ margin }>
      <li>Lorem ipsum dolor sit amet, consectetuer adipiscing elit.</li>
      <li>Aliquam tincidunt mauris eu risus.</li>
      <li>Vestibulum auctor dapibus neque.</li>
      <li>Nunc dignissim risus id metus.</li>
      <li>
        <p>Cras ornare tristique elit.</p>
      </li>
      <li>
        Praesent dapibus, neque id cursus faucibus, tortor neque egestas augue, eu vulputate magna
        eros eu erat. Aliquam erat volutpat. Nam dui mi, tincidunt quis, accumsan porttitor,
        facilisis luctus, metus.
      </li>
      <li>
        Phasellus ultrices nulla quis nibh. Quisque a lectus. Donec consectetuer ligula vulputate
        sem tristique cursus. Nam nulla quam, gravida non, commodo a, sodales sit amet, nisi.
      </li>
    </Ul>
  );
};

Basic.story = {
  decorators: [
    withKnobs,
    withTests({
      results
    })
  ],
  parameters: {
    jest: ['ul'],
    notes: README
  }
};

export default {
  title: 'Atoms/Typography/UL'
};