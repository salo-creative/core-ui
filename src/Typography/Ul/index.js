import Ul from './ul';

export default Ul;