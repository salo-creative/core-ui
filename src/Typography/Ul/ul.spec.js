import React from 'react';
import 'jest-styled-components';

// HELPERS
import { fluidTypeVals, renderWithTheme } from '../../../test';
import { colours } from '../../helpers/colours';

// COMPONENT
import Ul from './index';

test('Render default Ul tag and styles', async () => {
  // Variables
  const minFS = 14;
  const maxFS = 17;
  const minLH = 17;
  const maxLH = 29;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Ul data-testid='test-list'>
      <li>1 - Lorem ipsum</li>
      <li>2 - Lorem ipsum</li>
      <li>3 - Lorem ipsum</li>
    </Ul>
  );
  const el = getByTestId('test-list');
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  // Assert
  expect(el).toHaveStyleRule('color', colours.charcoal);
  expect(el).toHaveStyleRule('margin', '0 0 20px');
  expect(el).toHaveStyleRule('padding', '0 0 0 25px');
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});

test('Render Ul tag with core props', async () => {
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Ul
      data-testid='test-list'
      padding='10px'
      margin='0 0 10px'
      fontSize='25px'
      lineHeight='30px'
      color='#333'
    >
      <li>1 - Lorem ipsum</li>
      <li>2 - Lorem ipsum</li>
      <li>3 - Lorem ipsum</li>
    </Ul>
  );
  const el = getByTestId('test-list');
  // Assert
  expect(el).toHaveStyleRule('margin', '0 0 10px');
  expect(el).toHaveStyleRule('padding', '10px');
  expect(el).toHaveStyleRule('font-size', '25px');
  expect(el).toHaveStyleRule('line-height', '30px');
  expect(el).toHaveStyleRule('color', '#333');
  expect(container.firstChild).toMatchSnapshot();
});

test('Render Ul tag with fluid type props', async () => {
  // Variables
  const minFS = 10;
  const maxFS = 18;
  const minLH = 18;
  const maxLH = 26;
  // Render
  const { getByTestId, container } = renderWithTheme(
    <Ul
      data-testid='test-list'
      minFont={ minFS }
      maxFont={ maxFS }
      minLine={ minLH }
      maxLine={ maxLH }
    >
      <li>1 - Lorem ipsum</li>
      <li>2 - Lorem ipsum</li>
      <li>3 - Lorem ipsum</li>
    </Ul>
  );
  const el = getByTestId('test-list');
  // Assert
  const fontSizes = fluidTypeVals({ minLH, maxLH, maxFS, minFS });
  expect(el).toHaveStyleRule('font-size', `${ minFS }px`);
  expect(el).toHaveStyleRule('font-size', fontSizes.minFont, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('font-size', `${ maxFS }px`, { media: 'screen and (min-width:1200px)' });
  expect(el).toHaveStyleRule('line-height', `${ minLH }px`);
  expect(el).toHaveStyleRule('line-height', fontSizes.minLineHeight, { media: 'screen and (min-width:400px)' });
  expect(el).toHaveStyleRule('line-height', `${ maxLH }px`, { media: 'screen and (min-width:1200px)' });
  expect(container.firstChild).toMatchSnapshot();
});