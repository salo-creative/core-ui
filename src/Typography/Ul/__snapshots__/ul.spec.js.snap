// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Ul tag with core props 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #333;
  margin: 0 0 10px;
  padding: 10px;
  font-size: 25px;
  line-height: 30px;
}

.c0 li p {
  margin: 0;
}

<ul
  class="c0"
  color="#333"
  data-testid="test-list"
  font-size="25px"
>
  <li>
    1 - Lorem ipsum
  </li>
  <li>
    2 - Lorem ipsum
  </li>
  <li>
    3 - Lorem ipsum
  </li>
</ul>
`;

exports[`Render Ul tag with fluid type props 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #333333;
  margin: 0 0 20px;
  padding: 0 0 0 25px;
  font-size: 10px;
  line-height: 18px;
}

.c0 li p {
  margin: 0;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(10px + (18 - 10) * (100vw - 400px) / (1200 - 400));
    line-height: calc(18px + (26 - 18) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 18px;
    line-height: 26px;
  }
}

<ul
  class="c0"
  color=""
  data-testid="test-list"
  font-size=""
>
  <li>
    1 - Lorem ipsum
  </li>
  <li>
    2 - Lorem ipsum
  </li>
  <li>
    3 - Lorem ipsum
  </li>
</ul>
`;

exports[`Render default Ul tag and styles 1`] = `
.c0 {
  font-weight: 400;
  width: 100%;
  color: #333333;
  margin: 0 0 20px;
  padding: 0 0 0 25px;
  font-size: 14px;
  line-height: 17px;
}

.c0 li p {
  margin: 0;
}

@media screen and (min-width:400px) {
  .c0 {
    font-size: calc(14px + (17 - 14) * (100vw - 400px) / (1200 - 400));
    line-height: calc(17px + (29 - 17) * (100vw - 400px) / (1200 - 400));
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    font-size: 17px;
    line-height: 29px;
  }
}

<ul
  class="c0"
  color=""
  data-testid="test-list"
  font-size=""
>
  <li>
    1 - Lorem ipsum
  </li>
  <li>
    2 - Lorem ipsum
  </li>
  <li>
    3 - Lorem ipsum
  </li>
</ul>
`;
